google/cloud/bigquery/table.py

```diff
@@ -2071,6 +2121,24 @@ def to_dataframe(
         if string_dtype is not None and not hasattr(string_dtype, "__from_arrow__"):
             raise ValueError("string_dtype", _NO_SUPPORTED_DTYPE)
 
+        if (
+            date_dtype is not None
+            and date_dtype is not DefaultPandasDTypes.DATE_DTYPE
+            and not hasattr(date_dtype, "__from_arrow__")
+        ):
+            raise ValueError("date_dtype", _NO_SUPPORTED_DTYPE)
+
+        if datetime_dtype is not None and not hasattr(datetime_dtype, "__from_arrow__"):
+            raise ValueError("datetime_dtype", _NO_SUPPORTED_DTYPE)
+
+        if time_dtype is not None and not hasattr(time_dtype, "__from_arrow__"):
+            raise ValueError("time_dtype", _NO_SUPPORTED_DTYPE)
```

Comment:
Missing coverage for this line. Let's add a unit test for it.

https://source.cloud.google.com/results/invocations/339fcdd9-2b85-4e0c-b635-564fef1588c2/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-bigquery%2Fpresubmit%2Fpresubmit/log

Reply:
Thanks to catch it!